pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/defaultvalues/RgpAutomaticProbandDataCreator.java

```diff
@@ -37,8 +40,28 @@ protected boolean setDefaultData() {
                     fieldSettings.getColumnsWithDefaultOptionsFilteredByElasticExportWorkflow(fieldSettingsDtosByOptionAndInstanceId);
             String participantId = StringUtils.isNotBlank(esProfile.getLegacyAltPid()) ? esProfile.getLegacyAltPid() : esProfile.getGuid();
             ParticipantData participantData = new ParticipantData(participantDataDao);
-            Optional<BookmarkDto> maybeFamilyIdOfBookmark = bookmarkDao.getBookmarkByInstance(RGP_FAMILY_ID);
-            Map<String, String> probandDataMap = extractProbandDefaultDataFromParticipantProfile(maybeFamilyIdOfBookmark);
+
+            Optional<BookmarkDto> familyIdOfBookmark = bookmarkDao.getBookmarkByInstance(RGP_FAMILY_ID);
+            if (familyIdOfBookmark.isEmpty()) {
+                // nothing in the call stack will handle a throw properly, so log the error and use the return value
+                // to unwind
+                logger.error("RGP family ID not found in Bookmark table");
+                return false;
+            }
+            BookmarkDto familyIdBookmarkDto =  familyIdOfBookmark.get();
+
+            List<Activities> activities = elasticSearchParticipantDto.getActivities();
+            Map<String, String> probandDataMap =
+                    extractProbandDefaultData(activities, familyIdBookmarkDto.getValue());
+
+            try {
+                String refSourceId = convertReferralSources(getReferralSources(activities));
+                probandDataMap.put(REFERRAL_SOURCE_ID, refSourceId);
+            } catch (Exception e) {
+                // not good, but not fatal for this process
+                logger.error("Error deriving participant referral source for {}", participantId, e);
```

Comment:
Silly question.  What shows up in the log statement related to e?  I only see one substitution in this log statement and if I'm reading that correctly, that's going to be the participantId.